collectors/ebpf.plugin/ebpf_cgroup.c

```diff
@@ -56,40 +67,46 @@ void ebpf_map_cgroup_shared_memory()
     limit_try++;
     next_try = curr_time + NETDATA_EBPF_CGROUP_NEXT_TRY_SEC;
 
-    shm_fd_ebpf_cgroup = shm_open(NETDATA_SHARED_MEMORY_EBPF_CGROUP_NAME, O_RDWR, 0660);
     if (shm_fd_ebpf_cgroup < 0) {
-        if (limit_try == NETDATA_EBPF_CGROUP_MAX_TRIES)
-            error("Shared memory was not initialized, integration between processes won't happen.");
+        shm_fd_ebpf_cgroup = shm_open(NETDATA_SHARED_MEMORY_EBPF_CGROUP_NAME, O_RDWR, 0660);
+        if (shm_fd_ebpf_cgroup < 0) {
+            if (limit_try == NETDATA_EBPF_CGROUP_MAX_TRIES)
+                error("Shared memory was not initialized, integration between processes won't happen.");
 
-        return;
+            return;
+        }
     }
 
     // Map only header
-    shm_ebpf_cgroup.header = (netdata_ebpf_cgroup_shm_header_t *) ebpf_cgroup_map_shm_locally(shm_fd_ebpf_cgroup,
+    netdata_ebpf_cgroup_shm_header_t *header = (netdata_ebpf_cgroup_shm_header_t *) ebpf_cgroup_map_shm_locally(shm_fd_ebpf_cgroup,
                                                                                              sizeof(netdata_ebpf_cgroup_shm_header_t));
-    if (!shm_ebpf_cgroup.header) {
-        limit_try = NETDATA_EBPF_CGROUP_MAX_TRIES + 1;
+    if (!header) {
```

Comment:
This needs to be reviewed.As the `ebpf_cgroup_map_shm_locally` is using mmap. If mmap fails --> `On error, the value MAP_FAILED (that is, (void *) -1) is returned`. So to check for failure it needs to check again `MAP_FAILED`

Need to check all mmap functions that assume it would return NULL on failure

Reply:
Thank you, I fixed the issue with last commit.